### PR TITLE
fix(fdr2csv): refactor to unify fdr2csv, fix conversion bug

### PR DESCRIPTION
**Pull request by lukecologne**
## Summary of Changes

This PR fixes a bug in the new FDR2CSV, where a wrong type was used in the FdrData struct was used, leading to garbled data when decoded. It also refactors the FDR2CSV to autodetect the aircraft type and decode accordingly, improving usability.
Also adds an option `--get_raw_input_file_version`/`-r` to print out only the file version, for automation purposes, and makes `-o` optional when `-r` or `-g` are present
Further, fix a bug in the FDR itself, where a float value was written to the INI instead of a string of int.

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: tools/fdr2csv/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    buf_writer.write(header.as_bytes())?;
+
+    // Create the CSV writer, and serialize the file.
+    let mut writer = WriterBuilder::new()
+        .delimiter(args.delimiter as u8)
+        .has_headers(false)
+        .from_writer(buf_writer);
+
+    match aircraft_type {
+        AircraftType::A320 => {
```
Comment (lukecologne) [inline tools/fdr2csv/src/main.rs:145]: If any rustaceans have feedback on how to do this better, I'd love to hear it

Comment (pwnorbitals) [reply tools/fdr2csv/src/main.rs:145]: I think you might be able to define a `read_record_function` function pointer 
```
let read_record_function = match aircraft_type { AircraftType::A320 =>  {a320::read_record}, AircraftType::A380 => { a380::read_record } }
```
Then use that in your common workflow. Another approach could be to define a `read_record` function in your AircraftType that returns the right value depending on the variant.

Comment (lukecologne) [reply tools/fdr2csv/src/main.rs:145]: I tried defining the function as a variable, but it doesn't work as the return types of the functions are different

Comment (lukecologne) [reply tools/fdr2csv/src/main.rs:145]: And defining the return type as Box<dyn Serialize> also doesn't work:
```
    |
54  | pub fn read_record(reader: &mut impl Read) -> Result<Box<dyn Serialize>, Error> {
    |                                                          ^^^^^^^^^^^^^ `a320::_::_serde::Serialize` cannot be made into an object
    |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
```

**Review by aguther — APPROVED** [review]
- loaded up A32NX and A380X and created FDR files
- converted them using the fdr2csv
- checked the data for plausibility

Test succeeded. Approved.

**Pull request merged**